tools/generators/migrate-v8-pkg/index.ts

```diff
@@ -1,28 +1,51 @@
 import * as path from 'path';
-import { Tree, formatFiles, installPackagesTask, names, generateFiles } from '@nrwl/devkit';
-import { libraryGenerator } from '@nrwl/workspace/generators';
+import type { Linter } from 'eslint';
+import {
+  logger,
+  Tree,
+  formatFiles,
+  names,
+  readProjectConfiguration,
+  readJson,
+  joinPathFragments,
+  ProjectConfiguration,
+} from '@nrwl/devkit';
+import { fileExists } from '@nrwl/tao/src/utils/app-root';
 
-import { getProjectConfig } from '../../utils';
+import { getProjectConfig, getProjects } from '../../utils';
 
 import { MigrateV8PkgGeneratorSchema } from './schema';
+import { PackageJson, TsConfig } from '../../types';
+import { getCjsConfigObjectAst, getASTconfigObjectProp } from './lib/utils';
 
 interface NormalizedSchema extends ReturnType<typeof normalizeOptions> {}
+interface AssertedSchema extends MigrateV8PkgGeneratorSchema {
+  name: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
 
 export default async function (tree: Tree, schema: MigrateV8PkgGeneratorSchema) {
-  await libraryGenerator(tree, { name: schema.name });
+  const validatedSchema = await validateSchema(tree, schema);
+
+  if (hasSchemaFlag(validatedSchema, 'stats')) {
+    printStats(tree, validatedSchema);
 
-  const normalizedOptions = normalizeOptions(tree, schema);
+    return noop;
```

Comment:
Does nx require us to always return an "effects" function?

Reply:
nope, just unification of approach from my side